### PR TITLE
Clean up state files in temp directory when all uploads are done.

### PR DESCRIPTION
**Pull request by ua741**

### CODE REVIEW

File: lib/utils/file_uploader.dart [hunk 1]
```diff
@@ -282,6 +283,29 @@ class FileUploader {
     }
   }
 
+
+  Future<void> removeStaleFiles() async {
+    try {
+      final String dir = Configuration.instance.getTempDirectory();
```
Comment (vishnukvmd) [inline lib/utils/file_uploader.dart:289]: Would it make more sense to create a sub-dir for keeping these files and just removing that?

File: lib/services/remote_sync_service.dart [hunk 2]
```diff
@@ -139,6 +139,11 @@ class RemoteSyncService {
           Bus.instance.fire(SyncStatusUpdate(SyncStatus.completedBackup));
         }
       } else {
+        // if no files are uploaded and filesToBeUploaded is empty, clear
+        // stale files in the temp directory
+        if(!hasUploadedFiles && filesToBeUploaded.isEmpty) {
```
Comment (vishnukvmd) [inline lib/services/remote_sync_service.dart:144]: nit: lint

fwiw, I have auto-format configured to run on on file-save

File: lib/utils/file_uploader.dart (hunk 1 above)
Comment (ua741) [reply lib/utils/file_uploader.dart:289]: I initially thought of starting with that but the current configuration file is little too much and confusing and didn't want to refactor that yet. We will fix these when we pick resumable upload and download

File: lib/services/remote_sync_service.dart [hunk 3]
```diff
@@ -139,6 +139,11 @@ class RemoteSyncService {
           Bus.instance.fire(SyncStatusUpdate(SyncStatus.completedBackup));
         }
       } else {
+        // if no files are uploaded and filesToBeUploaded is empty, clear
+        // stale files in the temp directory
+        if (!hasUploadedFiles && filesToBeUploaded.isEmpty) {
```
Comment (vishnukvmd) [inline lib/services/remote_sync_service.dart:144]: Could just be `filesToBeUploaded.isEmpty`?

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**